syntax = "proto3";
package ProtobufMessage;

message MessagePackage
{
    int32 Function = 1;
    int32 PriorityLevel = 2;
    int32 SerialNumber = 3;
    string CheckString = 4;
    int32 Sender = 5;
    int32 Receiver = 6;
    string Context = 7;
    bytes ExtendData = 8;
}

message GLOBAL_CONFIG
{
    int32     ListenPort = 1;
    string  LogPath = 2;
    int32 LogLevel =3 ;
    bool    DebugMode = 4;
    int32     MQThreadQuantity =5;
    bool    EnableSSL=6;
    string  CertFilePath=7;
    string  KeyFilePath=8;
    string  CertPasswd=9;
    int32     ConnectTime=10;
    int32     ReceiveTime=11;
    int32     EPollThreadQuantity=12;
};

message DB_CONFIG
{
    string  ServerAddress=1;
    int32   ServerPort=2;
    string  User=3;
    string  Passwd=4;
    string  Database=5;
    GLOBAL_CONFIG   ServerConfig=6;
}

message DATA_CONFIG
{
    int32     DataReceivePort=1; 
    int32     DataRecvTime=2;
    GLOBAL_CONFIG   ServerConfig=3;
} 

message PROXY_CONFIG{
    int32     ClientCheckTime=1;
    string  ClientCheckKey=2;
    int32     TimeoutCheckInterval=3;
    int32       TimeoutTime=4;
    string      ProcessAddress=5;
    GLOBAL_CONFIG   ServerConfig=6;
};

message FIREWALL_CONFIG{
    GLOBAL_CONFIG   ServerConfig=5;
};

message PROCESS_CONFIG{
    int32  LuaContextQuantity=1;
    string  LuaScriptFolder=2;
    string  LuaEntryFile=3;
    string  LuaEntryFunction=4;
    string  LuaProcessFunction=5;
    string  LuaSocketCloseFunction=6;
    GLOBAL_CONFIG   ServerConfig=7;
};
